conn.go

```diff
@@ -195,6 +196,7 @@ func (conn *Conn) Close() error {
 		close(conn.eavesdropped)
 	}
 	conn.eavesdroppedLck.Unlock()
+	conn.outWait.Wait()
```

Comment:
We closed "conn.out" which is a channel. Here we wait on "conn.outWait" which is a WaitGroup independent of that channel.